exporter/elasticsearchexporter/internal/objmodel/objmodel.go

```diff
@@ -250,17 +250,28 @@ func (doc *Document) iterJSON(v *json.Visitor, dedot bool) error {
 }
 
 func (doc *Document) iterJSONFlat(w *json.Visitor) error {
```

Comment:
Name the return "argument" `err` and in the defer `func` do `err = w.OnObjectFinished()`, then instead of log.Fatal you will actually return an error.

Reply:
if this is done the error returned from the defers will trigger the errcheck lint check as the errors returned from the defers are not checked

These are some related issues found 
https://github.com/kisielk/errcheck/issues/101
https://github.com/kubernetes-sigs/cluster-api/issues/4586

Reply:
I don't think so, can you show me?

```golang
func (doc *Document) iterJSONFlat(w *json.Visitor) (err error) {
	err = w.OnObjectStart(-1, structform.AnyType)
	if err != nil {
		return 
	}
	defer func() {
		err = w.OnObjectFinished()
	}()
        ....
}

```

Reply:
done like this 
```
defer func() error {
		err = w.OnObjectFinished()
		if err != nil {
			return err
		}
		return nil
	}()
```
getting this 
![Selection_005](https://user-images.githubusercontent.com/45638240/171342168-ab06c8e5-a8c3-438b-9cd4-75fc64d6f522.png)

Reply:
@DiptoChakrabarty, please take a closer look at Bogdan's example. You cannot return the error from the deferred function, but you can assign the error to a return value on the calling function.